cpp/src/barretenberg/honk/composer/composer_helper/standard_plonk_composer_helper.cpp

```diff
@@ -87,21 +42,38 @@ void StandardPlonkComposerHelper<CircuitConstructor>::compute_witness(const Circ
 }
 
 /**
- * Compute proving key.
- * Compute the polynomials q_l, q_r, etc. and sigma polynomial.
+ * Compute proving key
+ *
+ * 1. Load crs.
+ * 2. Initialize this.circuit_proving_key.
+ * 3. Create constraint selector polynomials from each of this composer's `selectors` vectors and add them to the
+ * proving key.
+ * 4. Compute sigma polynomial
  *
- * @return Proving key with saved computed polynomials.
+ * @return Pointer to the initialized proving key updated with selector polynomials.
  * */
-
 template <typename CircuitConstructor>
 std::shared_ptr<bonk::proving_key> StandardPlonkComposerHelper<CircuitConstructor>::compute_proving_key(
     const CircuitConstructor& circuit_constructor)
 {
     if (circuit_proving_key) {
         return circuit_proving_key;
     }
-    // Compute q_l, q_r, q_o, etc polynomials
-    StandardPlonkComposerHelper::compute_proving_key_base(circuit_constructor, plonk::ComposerType::STANDARD_HONK);
+    const size_t minimum_circuit_size = 0;
+    const size_t num_randomized_gates = NUM_RANDOMIZED_GATES;
+    // Initialize circuit_proving_key
+    // TODO(#229)(Kesha): replace composer types.
+    circuit_proving_key = initialize_proving_key(circuit_constructor,
+                                                 crs_factory_.get(),
+                                                 minimum_circuit_size,
+                                                 num_randomized_gates,
+                                                 plonk::ComposerType::STANDARD);
+    // Compute lagrange selectors
+    construct_lagrange_selector_forms(circuit_constructor, circuit_proving_key.get());
```

Comment:
This is an easy change but it would touch a lot of files. This pattern exists in many places throughout the code and would be better handled all at once as one simple PR.

Reply:
Added issue and a TODO